cin_validator/rules/cin2022_23/rule_8730.py

```diff
@@ -0,0 +1,71 @@
+from typing import Mapping
+
+import pandas as pd
+
+from cin_validator.rule_engine import rule_definition, CINTable, RuleContext
+from cin_validator.rule_engine import IssueLocator
+from cin_validator.test_engine import run_rule
+
+# Get tables and columns of interest from the CINTable object defined in rule_engine/__api.py
+# Replace ChildIdentifiers with the table name, and LAChildID with the column name you want.
+
+ChildProtectionPlans = CINTable.ChildProtectionPlans
+NumberOfPreviousCPP = ChildProtectionPlans.NumberOfPreviousCPP
+
+# define characteristics of rule
+@rule_definition(
+    # write the rule code here, in place of 8500
+    code=8730,
+    # replace ChildIdentifiers with the value in the module column of the excel sheet corresponding to this rule .
+    module=CINTable.ChildProtectionPlans,
+    # replace the message with the corresponding value for this rule, gotten from the excel sheet.
+    message="Total Number of previous Child Protection Plans missing",
+    # The column names tend to be the words within the < > signs in the github issue description.
+    affected_fields=[NumberOfPreviousCPP],
+)
+def validate(
+    data_container: Mapping[CINTable, pd.DataFrame], rule_context: RuleContext
+):
+    # Replace ChildIdentifiers with the name of the table you need.
+    df = data_container[ChildProtectionPlans]
+
+    # implement rule logic as described by the Github issue. Put the description as a comment above the implementation as shown.
+
+    # Where a CPP module is present, <NumberOfPreviousCPP> (N00106) must be greater than or equal to zero
+    # Change the line below to ensure values are >=0 ie not null
+
+    # <LAchildID> (N00097) must be present
+    failing_indices = df[df[NumberOfPreviousCPP].isna()].index
```

Comment:
This is good, I didn't initially think of it along these lines.

Sidenote:
Are we assuming that negative values can never occur?
If they do, they'll pass this condition. 

It might be useful to check filter missing values or values that are less than zero.